crates/bevy_core_pipeline/src/bloom/mod.rs

```diff
@@ -576,180 +370,122 @@ fn prepare_bloom_textures(
-                    layout: &pipelines.upsampling_bind_group_layout,
-                    entries: &[
-                        BindGroupEntry {
-                            binding: 0,
-                            resource: BindingResource::TextureView(&org),
-                        },
-                        BindGroupEntry {
-                            binding: 1,
-                            resource: BindingResource::Sampler(&pipelines.sampler),
-                        },
-                        BindGroupEntry {
-                            binding: 2,
-                            resource: uniforms.clone(),
-                        },
-                        BindGroupEntry {
-                            binding: 3,
-                            resource: BindingResource::TextureView(&up),
-                        },
-                    ],
-                });
-
-                upsampling_bind_groups.push(bind_group);
-            }
-
-            let upsampling_final_bind_group =
-                render_device.create_bind_group(&BindGroupDescriptor {
-                    label: Some("bloom_upsampling_final_bind_group"),
-                    layout: &pipelines.downsampling_bind_group_layout,
-                    entries: &[
-                        BindGroupEntry {
-                            binding: 0,
-                            resource: BindingResource::TextureView(&BloomTextures::texture_view(
-                                &textures.texture_b,
-                                0,
-                            )),
-                        },
-                        BindGroupEntry {
-                            binding: 1,
-                            resource: BindingResource::Sampler(&pipelines.sampler),
-                        },
-                        BindGroupEntry {
-                            binding: 2,
-                            resource: uniforms.clone(),
-                        },
-                    ],
-                });
-
-            commands.entity(entity).insert(BloomBindGroups {
-                prefilter_bind_group,
-                downsampling_bind_groups: downsampling_bind_groups.into_boxed_slice(),
-                upsampling_bind_groups: upsampling_bind_groups.into_boxed_slice(),
-                upsampling_final_bind_group,
-            });
+                    BindGroupEntry {
+                        binding: 1,
+                        resource: BindingResource::Sampler(&sampler),
+                    },
+                ],
+            }));
         }
+
+        commands.entity(entity).insert(BloomBindGroups {
+            downsampling_bind_groups: downsampling_bind_groups.into_boxed_slice(),
+            upsampling_bind_groups: upsampling_bind_groups.into_boxed_slice(),
+            sampler,
+        });
     }
 }
 
-fn calculate_mip_count(min_view: u32) -> u32 {
-    ((min_view as f32).log2().round() as i32 - 3).max(1) as u32
+impl BloomSettings {
```

Comment:
I feel like this should be with the type definition. Having the impl here and the type definition in another file is unexpected.

Reply:
This function is only used in this file. It's mostly implemented as a method for ease of use, it could be a standalone helper function. I'd rather not put it with the settings definition and methods people will actually use in the other file.

Reply:
Ok, please make it a plain function that takes a &BloomSettings instead of a method.